File: ios/Poste_iOS/Poste_iOS/Network/DTO/PagedResponse.swift
import Foundation


struct PagedResponse<T: Codable>: Codable {
    
    let data: [T]
    let nextCursor: String?
    let links: PageMeta
    
    struct PageMeta: Codable {
        
        let size: Int?
        let nextCursor: String?
        
    }
    
}

File: multiplatform/multiPlatform/composeApp/src/commonMain/kotlin/network/dto/PagedResponse.kt
package network.dto

import kotlinx.serialization.Serializable

/**
 * Generic wrapper for paginated responses
 */
@Serializable
data class PagedResponse<T>(
    val data: List<T>,
    val page: PageMeta
) {
    @Serializable
    data class PageMeta(
        val size: Int? = null,
        val nextCursor: String? = null
    )
}

